Makefile

```diff
@@ -3,16 +3,16 @@ EXTVERSION = 0.1.0
 
 MODULE_big = embedding
 DATA = $(wildcard *--*.sql)
-OBJS = embedding.o hnswalg.o
+OBJS = embedding.o hnswalg.o distfunc.o
 
 TESTS = $(wildcard test/sql/*.sql)
 REGRESS = $(patsubst test/sql/%.sql,%,$(TESTS))
 REGRESS_OPTS = --inputdir=test --load-extension=embedding
 
 # For auto-vectorization:
-# - GCC (needs -ftree-vectorize OR -O3) - https://gcc.gnu.org/projects/tree-ssa/vectorization.html
-PG_CFLAGS += -O3
-PG_CXXFLAGS += -O3 -std=c++11
+# - GCC&clang needs -Ofast or -O3: https://gcc.gnu.org/projects/tree-ssa/vectorization.html
+PG_CFLAGS += -Ofast -mavx2
```

Comment:
I don't think we should put `-mavx2` here, I think distance functions should have the dynamic dispatch thing as before. If it's possible you can recompile `distfunc.c` twice, once with AVX2 and once without. Otherwise you can copy-paste each function twice and mark it with `__attribute__((target("avx2")))`

Reply:
Also, try enabling `fma` along with `avx2`, I think it should give a small speedup

Reply:
Restored code with intrinsics